subsys/spm/spm.c

```diff
@@ -567,3 +582,93 @@ void spm_config(void)
 	spm_config_sram();
 	spm_config_peripherals();
 }
+
+#if CONFIG_SPM_NS_DEBUG_LOGS_DUMP
+
+static spm_ns_on_fatal_error_t ns_on_fatal_handler;
+
+__TZ_NONSECURE_ENTRY_FUNC
+int spm_set_ns_fatal_error_handler(spm_ns_on_fatal_error_t handler)
+{
+	ns_on_fatal_handler = handler;
+
+	return 0;
+}
+
+void spm_ns_fatal_error_handler(void)
+{
+	TZ_NONSECURE_FUNC_PTR_DECLARE(fatal_handler_ns);
+	fatal_handler_ns = TZ_NONSECURE_FUNC_PTR_CREATE(ns_on_fatal_handler);
+
+	if (TZ_NONSECURE_FUNC_PTR_IS_NS(fatal_handler_ns)) {
+		fatal_handler_ns();
+	}
+}
+#endif /* CONFIG_SPM_NS_DEBUG_LOGS_DUMP */
+
+#if CONFIG_SPM_SHARE_CONSOLE_UART
```

Comment:
This section might merit being placed in its own separate file. What do you think?